Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs

```diff
@@ -271,6 +279,18 @@ public ActionResult BulkDeleteSubmissions([DataSourceRequest]DataSourceRequest r
             }
 
             this.Data.SaveChanges();
+            
+            foreach (GridModelType submission in submissions)
+            {
+                var dbSubmission = this.Data.Submissions.GetById(submission.Id);
+
+                if (dbSubmission.ParticipantId.HasValue && dbSubmission.ProblemId.HasValue)
+                {
+                    this.Data.ParticipantScores.RecalculateParticipantScore(dbSubmission.ParticipantId.Value, dbSubmission.ProblemId.Value);
+                }
+            }
+
+            this.Data.SaveChanges();
```

Comment:
An action's DB modification should be executed as a single transaction, to ensure no invalid state in the DB can be reached